cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -112,20 +112,11 @@ protected int getInstances(HttpRequest request) throws BadRequestException {
   }
 
   protected Map<String, String> decodeArguments(HttpRequest request) throws JsonSyntaxException {
-    ChannelBuffer content = request.getContent();
-    if (!content.readable()) {
+    Map<String, String> args = parseBody(request, STRING_MAP_TYPE);
```

Comment:
nice cleanup. So the only thing that we lose because of this is the message? Worth it to have ``parseBody`` accept an error message too?